src/sensors/tests/test_distant.py

```diff
@@ -218,8 +228,38 @@ def test_render(variant_scalar_rgb):
         "integrator": {"type": "path"}
     }
 
+    # set the sensor origin such that rays hit the square
+    if not ray_origin:
+        pass
+    elif ray_origin["ray_origin"] == "disk":
+        ray_origin["ray_origin_center"] = [-w_o[i]*2 for i in range(3)]
+    elif ray_origin["ray_origin"] == "rectangle":
+        ray_origin["ray_origin_a"] = [-1-w_o[0]*2, -1-w_o[1]*2, 0-w_o[2]*2]
+        ray_origin["ray_origin_b"] = [1-w_o[0]*2, 1-w_o[1]*2, 0-w_o[2]*2]
+
+    dict_scene["sensor"] = {**dict_scene["sensor"], **ray_origin}
     scene = load_dict(dict_scene)
+
+    # if the ray origin area does not exactly cover the surface in the scene,
+    # the expected recorded radiance has to be corrected
+    if not ray_origin:
+        ray_origin_area = 2 * np.pi
+    elif ray_origin["ray_origin"] == "disk":
+        ray_origin_area = (ray_origin["ray_origin_radius"]**2) * np.pi
+    elif ray_origin["ray_origin"] =="rectangle":
+        ray_origin_a = ray_origin["ray_origin_a"]
+        ray_origin_b = ray_origin["ray_origin_b"]
+        ray_origin_area = abs(ray_origin_a[0] - ray_origin_b[0]) *  abs(ray_origin_a[1] - ray_origin_b[1])
+    
+    if ray_origin_area >= surface_area:
+        ratio = surface_area / ray_origin_area
+    else:
+        ratio = ray_origin_area / surface_area
+
     sensor = scene.sensors()[0]
     scene.integrator().render(scene, sensor)
     img = np.array(sensor.film().bitmap()).squeeze()
-    assert np.allclose(np.array(img), expected)
+    assert np.allclose(np.array(img), expected * ratio, rtol=1e-3)
```

Comment:
I suppose I am testing a different thing than what you have in mind.

My idea is this: On one hand, we might not be able to match the surface and the origin shape completely, e.g if we have a rectangular surface and a circular origin area. On the other: We might not want to match them exactly. Fox example when we have a tiled surface to avoid boundary effects, we might want to target only an area in the middle of the entire scene.
I think it makes sense to test if the sensor records the correct fraction of radiance in that case.

I am not sure, what you mean by the check for invalid intersections, though. As we said yesterday: A ray missing the surface will contribute a sample of value 0, so if only half of all rays intersect with the surface, I will record half the radiance.

Reply:
Interesting enough, when I remove the result scaling (the `ratio`) the tests for the rectangular origin, which matches the surface pass, as do the tests with the `Distant` sampling. (Only with SPP of a factor 100 higher though) I am not sure about the latter, because here some rays miss the surface and I am unsure how they are accounted for. And if they are, why it does not work the same for the circular origin when it is larger than the surface :thinking:

Reply:
What I mean is that the shape you sample ray origins from should be such that all fired rays land on the reflecting surface. If this condition is satisfied, the result should be the same, whatever the selected origin sampling policy (rectangle or disk). And I expect the distant policy to yield wrong results.

> I think it makes sense to test if the sensor records the correct fraction of radiance in that case.

If you target a subset of the reflecting surface, you will record the same value as if you target it entirely (if the illumination is uniform, of course). Only if you get light (or an absence of it) from somewhere else will you get a different result.

> A ray missing the surface will contribute a sample of value 0, so if only half of all rays intersect with the surface, I will record half the radiance.

Indeed. This means that the invalid intersection test should be good enough to not re-test that here.